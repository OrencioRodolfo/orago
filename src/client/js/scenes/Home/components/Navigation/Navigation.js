import React, { Component } from 'react';
import styled from 'styled-components';
import { Link } from 'react-router-dom';
import RoundedContainer from '../../../../components/Containers/Rounded/Rounded';
import Icon from '../../../../components/Icon/Icon';

const List = styled.ul`
  margin: 0;
  padding: 0;
  display: flex;
`;

const Item = styled.li`
  display: inline-block;
  margin: 0;
  padding: 10px;
`;

const Stat = styled.span` 
  font-weight: ${props => props.theme.weight('semiBold')};
  margin: 0 5px;
`;

class Navigation extends Component {
  render() {
    return (
      <RoundedContainer>
        <List>
          <Item>
            <Link to="/commits">
              <Icon icon="clock-o" />
              <Stat>21</Stat>
              commits
            </Link>
          </Item>
          <Item>
            <Link to="/organizations">
              <Icon icon="users" />
              <Stat>3</Stat>
              organizations
            </Link>
          </Item>
          <Item>
            <Link to="/repos">
              <Icon icon="github-alt" />
              <Stat>5</Stat>
              repos
            </Link>
          </Item>
        </List>
      </RoundedContainer>
    );
  }
}

export default Navigation;
