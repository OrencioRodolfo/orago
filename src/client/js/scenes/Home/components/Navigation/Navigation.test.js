import React from 'react';
import { shallow } from 'enzyme';
import { Navigation, mapStateToProps } from './Navigation';

describe('Navigation component', () => {
  it('Must render a navigation with the totals', () => {
    const container = shallow(
      <Navigation totalCommits={31} />,
    );

    expect(container).toMatchSnapshot();
  });

  describe('mapStateToProps function', () => {
    it('Must return an object with no totals', () => {
      const defaultState = {
        history: {
          commits: {},
        },
      };
      const result = mapStateToProps(defaultState);
      const expected = {
        totalCommits: 0,
      };
      expect(result).toEqual(expected);
    });
    it('Must return an object with 6 commits as total', () => {
      const state = {
        history: {
          commits: {
            2015: [
              {},
              {},
              {},
            ],
            2016: [
              {},
            ],
            2017: [
              {},
              {},
            ],
          },
        },
      };
      const result = mapStateToProps(state);
      const expected = {
        totalCommits: 6,
      };
      expect(result).toEqual(expected);
    });
  });
});
