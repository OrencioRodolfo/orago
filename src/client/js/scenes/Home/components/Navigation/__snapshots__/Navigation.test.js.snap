// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation component Must render a navigation with the totals 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Navigation
    totalCommits={31}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "class",
    "props": Object {
      "children": <styled.ul>
        <styled.li>
          <Link
            className="active"
            onClick={[Function]}
            replace={false}
            to="/organizations"
          >
            <Icon
              icon="users"
            />
            <styled.span>
              3
            </styled.span>
            organizations
          </Link>
        </styled.li>
        <styled.li>
          <Link
            className=""
            onClick={[Function]}
            replace={false}
            to="/repos"
          >
            <Icon
              icon="github-alt"
            />
            <styled.span>
              5
            </styled.span>
            repos
          </Link>
        </styled.li>
      </styled.ul>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <styled.li>
            <Link
              className="active"
              onClick={[Function]}
              replace={false}
              to="/organizations"
            >
              <Icon
                icon="users"
              />
              <styled.span>
                3
              </styled.span>
              organizations
            </Link>
          </styled.li>,
          <styled.li>
            <Link
              className=""
              onClick={[Function]}
              replace={false}
              to="/repos"
            >
              <Icon
                icon="github-alt"
              />
              <styled.span>
                5
              </styled.span>
              repos
            </Link>
          </styled.li>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "/organizations",
          "nodeType": "class",
          "props": Object {
            "children": <Link
              className="active"
              onClick={[Function]}
              replace={false}
              to="/organizations"
            >
              <Icon
                icon="users"
              />
              <styled.span>
                3
              </styled.span>
              organizations
            </Link>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Icon
                  icon="users"
                />,
                <styled.span>
                  3
                </styled.span>,
                "organizations",
              ],
              "className": "active",
              "onClick": [Function],
              "replace": false,
              "to": "/organizations",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": null,
                "nodeType": "function",
                "props": Object {
                  "icon": "users",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "class",
                "props": Object {
                  "children": 3,
                },
                "ref": null,
                "rendered": 3,
                "type": [Function],
              },
              "organizations",
            ],
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "/repos",
          "nodeType": "class",
          "props": Object {
            "children": <Link
              className=""
              onClick={[Function]}
              replace={false}
              to="/repos"
            >
              <Icon
                icon="github-alt"
              />
              <styled.span>
                5
              </styled.span>
              repos
            </Link>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": null,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                <Icon
                  icon="github-alt"
                />,
                <styled.span>
                  5
                </styled.span>,
                "repos",
              ],
              "className": "",
              "onClick": [Function],
              "replace": false,
              "to": "/repos",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": null,
                "nodeType": "function",
                "props": Object {
                  "icon": "github-alt",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "class",
                "props": Object {
                  "children": 5,
                },
                "ref": null,
                "rendered": 5,
                "type": [Function],
              },
              "repos",
            ],
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "class",
      "props": Object {
        "children": <styled.ul>
          <styled.li>
            <Link
              className="active"
              onClick={[Function]}
              replace={false}
              to="/organizations"
            >
              <Icon
                icon="users"
              />
              <styled.span>
                3
              </styled.span>
              organizations
            </Link>
          </styled.li>
          <styled.li>
            <Link
              className=""
              onClick={[Function]}
              replace={false}
              to="/repos"
            >
              <Icon
                icon="github-alt"
              />
              <styled.span>
                5
              </styled.span>
              repos
            </Link>
          </styled.li>
        </styled.ul>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <styled.li>
              <Link
                className="active"
                onClick={[Function]}
                replace={false}
                to="/organizations"
              >
                <Icon
                  icon="users"
                />
                <styled.span>
                  3
                </styled.span>
                organizations
              </Link>
            </styled.li>,
            <styled.li>
              <Link
                className=""
                onClick={[Function]}
                replace={false}
                to="/repos"
              >
                <Icon
                  icon="github-alt"
                />
                <styled.span>
                  5
                </styled.span>
                repos
              </Link>
            </styled.li>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "/organizations",
            "nodeType": "class",
            "props": Object {
              "children": <Link
                className="active"
                onClick={[Function]}
                replace={false}
                to="/organizations"
              >
                <Icon
                  icon="users"
                />
                <styled.span>
                  3
                </styled.span>
                organizations
              </Link>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <Icon
                    icon="users"
                  />,
                  <styled.span>
                    3
                  </styled.span>,
                  "organizations",
                ],
                "className": "active",
                "onClick": [Function],
                "replace": false,
                "to": "/organizations",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "function",
                  "props": Object {
                    "icon": "users",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "class",
                  "props": Object {
                    "children": 3,
                  },
                  "ref": null,
                  "rendered": 3,
                  "type": [Function],
                },
                "organizations",
              ],
              "type": [Function],
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "/repos",
            "nodeType": "class",
            "props": Object {
              "children": <Link
                className=""
                onClick={[Function]}
                replace={false}
                to="/repos"
              >
                <Icon
                  icon="github-alt"
                />
                <styled.span>
                  5
                </styled.span>
                repos
              </Link>,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": null,
              "nodeType": "class",
              "props": Object {
                "children": Array [
                  <Icon
                    icon="github-alt"
                  />,
                  <styled.span>
                    5
                  </styled.span>,
                  "repos",
                ],
                "className": "",
                "onClick": [Function],
                "replace": false,
                "to": "/repos",
              },
              "ref": null,
              "rendered": Array [
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "function",
                  "props": Object {
                    "icon": "github-alt",
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                Object {
                  "instance": null,
                  "key": null,
                  "nodeType": "class",
                  "props": Object {
                    "children": 5,
                  },
                  "ref": null,
                  "rendered": 5,
                  "type": [Function],
                },
                "repos",
              ],
              "type": [Function],
            },
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
